src/react.tsx

```diff
@@ -31,7 +47,19 @@ function normalizeQuery<STORE extends typeof MSTGQLStore.Type, DATA>(
 ): Query<DATA> {
   if (typeof query === "function") return query(store)
   if (query instanceof Query) return query
-  return store.query(query, variables, { fetchPolicy, raw })
+  if (typeof window === "undefined") {
+    const queryPromise = store.query<DATA>(query, variables, {
+      fetchPolicy: fetchPolicy,
+      raw: raw
+    })
+    store.pushPromise(queryPromise.promise)
```

Comment:
to not leak memory, I think there should also be an `query.finally(() => { store.unpushPromise(queryPromise.promise) }` (cleanPromises is then no longer needed)

Reply:
I'll think we should perhaps move this logic to `Query.ts`?

Reply:
how do I unpush a promise?